File: blog-server/src/module/home/comment/model/comment.param.ts
import {IsInt, IsOptional, IsString} from 'class-validator'
import {PageParam} from "../../../../model/BaseParam";




export class CommentInsertParam {
    @IsString({ message: 'articleId必须是字符类型' })
    articleId: string

    @IsString({ message: 'content必须要是字符类型' })
    content: string

    @IsString({ message: 'reviewerName必须要是字符类型' })
    reviewerName: string

    @IsString({ message: 'qq必须要是字符类型' })
    qq: string

    @IsString()
    articleUrl: string

    @IsOptional()       // 表示 search 字段是可选的
    @IsString()
    replyQQ?: string

    @IsOptional()       // 表示 search 字段是可选的
    @IsInt()
    parentId?: number

    @IsOptional()       // 表示 search 字段是可选的
    @IsString()
    httpsrc?: string

    @IsOptional()       // 表示 search 字段是可选的
    @IsString()
    replyName?: string
}



export class CommentHomeFindParam extends PageParam {
    @IsString()
    articleId: string
}

File: BlogServer/Blog.Model/Param/CommentParam.cs

namespace Blog.Model.Param
{
    public class CommentFindParam: PageParam
    {
        public string? ArticleName {  get; set; }

        public string? Content { get; set; }
    }

    public class CommentUpdateParam 
    {
        public int Id { get; set; }
        public required string Content { get; set; }
    }

    public class CommentInsertParam
    { 
        public required string ArticleId { get; set; }
        public required string ReviewerName { get;set; }
        public required string Qq { get; set; }
        public required string Content { get; set; }
        public required string ArticleUrl { get; set; }
        public string? ReplyQQ { get; set; }
        public int? ParentId { get; set; }
        public string? Httpsrc { get; set; }
        public string? ReplyName { get; set; }
    }
    public class CommentHomeFindParam : PageParam
    {
        public required string ArticleId { get; set; }
    }
}
